src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -1231,4 +1225,30 @@ boolean isStarted() {
 
     private static final SimpleState INITIAL = new SimpleState(INITIAL_ID, false, false);
     private static final SimpleState FAILED = new SimpleState(FAILED_ID, true, true);
+
+    /**
+     * Determine if the given evidence can be trusted in lieu of verifying or acquiring a credential.
+     *
+     * @param realmIdentity the current realm identity
+     * @param domain the current domain
+     * @param evidence the evidence to check (must not be {@code null})
+     * @return {@code true} if the given evidence can be trusted, {@code false} otherwise
+     * @throws RealmUnavailableException if the realm is not able to handle requests for any reason
+     */
+    private static boolean checkEvidenceTrusted(final RealmIdentity realmIdentity, final SecurityDomain domain, final Evidence evidence) throws RealmUnavailableException {
```

Comment:
Since this code is referenced from a static nested class, its access should be package-private, not private.  If private, it'll still compile but the compiler has to generate an invisible "access" method with package-private access in order to make it work (which also shows up on stack traces and debuggers as a weird method call whose line number typically corresponds to the line number of the class).  Again not a big deal, just a possible cleanup item for later.